src/Catalog/Registration/RegistrationCollector.cs

```diff
@@ -57,7 +57,7 @@ protected override Task<IEnumerable<CatalogItemBatch>> CreateBatches(IEnumerable
                 .GroupBy(item => GetKey(item.Value))
                 .Select(group => new CatalogItemBatch(
                     group.Min(item => item.CommitTimeStamp),
-                    group));
+                    group.OrderBy(x => x.CommitTimeStamp)));
```

Comment:
I'm trying to understand why we do the chronological sort here instead of in `SortingCollector`. The name of `SortingCollector` suggests to me that it would be a better place.